core/src/main/java/io/confluent/kafka/schemaregistry/storage/InMemoryCache.java

```diff
@@ -255,48 +258,10 @@ public void clearSubjects(Predicate<String> match) {
     return s -> subject == null || subject.equals(s);
   }
 
-  // Visible for testing
-  protected void replaceMatchingDeletedWithNonDeletedOrRemove(Predicate<String> match) {
-    Predicate<SchemaKey> matchDeleted = matchDeleted(match);
-
-    // Iterate through the entries, and for each entry that matches and is soft deleted,
-    // see if there is a replacement entry (that is not soft deleted) that has the same
-    // schema string.  If so, replace, else remove the entry.
-    Iterator<Map.Entry<Integer, SchemaKey>> it = guidToSchemaKey.entrySet().iterator();
-    while (it.hasNext()) {
-      Map.Entry<Integer, SchemaKey> entry = it.next();
-      SchemaKey schemaKey = entry.getValue();
-      if (matchDeleted.test(schemaKey)) {
-        SchemaValue schemaValue = (SchemaValue) store.get(schemaKey);
-        // The value returned from the store should not be null since we clean up caches
-        // after tombstoning, but we still check defensively
-        SchemaKey newSchemaKey = schemaValue != null
-                                 ? getNonDeletedSchemaKey(schemaValue)
-                                 : null;
-        if (newSchemaKey != null) {
-          entry.setValue(newSchemaKey);
-        } else {
-          it.remove();
-        }
-      }
-    }
-  }
-
-  private SchemaKey getNonDeletedSchemaKey(SchemaValue schema) {
-    MD5 md5 = MD5.ofString(schema.getSchema(), schema.getReferences());
-    SchemaIdAndSubjects keys = schemaHashToGuid.get(md5);
-    return keys == null ? null : keys.findAny(key -> {
-      SchemaValue value = (SchemaValue) store.get(key);
-      // The value returned from the store should not be null since we clean up caches
-      // after tombstoning, but we still check defensively
-      return value != null && !value.isDeleted();
-    });
-  }
-
-  private Predicate<SchemaKey> matchDeleted(Predicate<String> match) {
-    return key -> {
-      if (match.test(key.getSubject())) {
-        SchemaValue value = (SchemaValue) store.get(key);
+  private BiPredicate<String, Integer> matchDeleted(Predicate<String> match) {
+    return (subject, version) -> {
+      if (match.test(subject)) {
+        SchemaValue value = (SchemaValue) store.get(new SchemaKey(subject, version));
         // The value returned from the store should not be null since we clean up caches
```

Comment:
These complicated methods were used to maintain `(guid -> schemaKey)` and are no longer needed